spec/models/billable_metric_spec.rb

```diff
@@ -28,4 +28,59 @@
       end
     end
   end
+
+  describe '#groups_as_tree' do
+    let(:billable_metric) { create(:billable_metric) }
+
+    context 'without active groups' do
+      it 'returns {}' do
+        expect(billable_metric.groups_as_tree).to eq({})
+      end
+    end
+
+    context 'when groups contain one dimension' do
+      before do
+        create(:group, billable_metric: billable_metric, key: 'country', value: 'france')
+        create(:group, billable_metric: billable_metric, key: 'country', value: 'italy')
+      end
+
+      it 'returns a tree with one dimension' do
+        expect(billable_metric.groups_as_tree).to eq(
+          {
+            key: 'country',
+            values: %w[france italy]
+          }
+        )
+      end
+    end
+
+    context 'when groups contain two dimensions' do
+      before do
+        france = create(:group, billable_metric: billable_metric, key: 'country', value: 'france')
+        italy = create(:group, billable_metric: billable_metric, key: 'country', value: 'italy')
+        create(:group, billable_metric: billable_metric, key: 'cloud', value: 'aws', parent_group_id: france.id)
+        create(:group, billable_metric: billable_metric, key: 'cloud', value: 'google', parent_group_id: france.id)
+        create(:group, billable_metric: billable_metric, key: 'cloud', value: 'google', parent_group_id: italy.id)
+      end
+
+      it 'returns a tree with two dimensions' do
+        expect(billable_metric.groups_as_tree).to eq(
+          {
+            key: 'country',
+            values: [
+              {
+                name: 'france',
+                key: 'cloud',
+                values: %w[aws google]
```

Comment:
Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.